contrib/platform/test/com/sun/jna/platform/win32/Advapi32Test.java

```diff
@@ -183,6 +183,70 @@ public void testOpenThreadOrProcessToken() {
     	assertTrue(Kernel32.INSTANCE.CloseHandle(phToken.getValue()));
     }
     
+    public void testSetThreadTokenCurrentThread() {
+    	HANDLEByReference phToken = new HANDLEByReference();
+    	HANDLEByReference phTokenDup = new HANDLEByReference();
+    	HANDLE threadHandle = Kernel32.INSTANCE.GetCurrentThread();
+    	// See if thread has a token. If not, must duplicate process token and set thread token using that.
+    	if (!Advapi32.INSTANCE.OpenThreadToken(threadHandle, 
+             WinNT.TOKEN_IMPERSONATE | WinNT.TOKEN_QUERY, false, phToken)) {
+        	assertEquals(W32Errors.ERROR_NO_TOKEN, Kernel32.INSTANCE.GetLastError());
+            HANDLE processHandle = Kernel32.INSTANCE.GetCurrentProcess();
+            assertTrue(Advapi32.INSTANCE.OpenProcessToken(processHandle, WinNT.TOKEN_DUPLICATE, phToken));
+            assertTrue(Advapi32.INSTANCE.DuplicateTokenEx(phToken.getValue(), 
+                        WinNT.TOKEN_IMPERSONATE,
+                        null,
+                        WinNT.SECURITY_IMPERSONATION_LEVEL.SecurityImpersonation,
+                        WinNT.TOKEN_TYPE.TokenImpersonation,
+                        phTokenDup));
+        	// Null sets on current thread
+            assertTrue(Advapi32.INSTANCE.SetThreadToken(null, phTokenDup.getValue()));        	
+    	}
+    	else
+    	{
+            //Null sets on current thread
+            assertTrue(Advapi32.INSTANCE.SetThreadToken(null, phToken.getValue()));
+    	}    	
+    	// Revert and cleanup
+    	assertTrue(Advapi32.INSTANCE.SetThreadToken(null, null));
+    	assertTrue(Kernel32.INSTANCE.CloseHandle(phToken.getValue()));
+    	if (phTokenDup.getValue() != null)
+            assertTrue(Kernel32.INSTANCE.CloseHandle(phTokenDup.getValue()));    	
+    }
+    
+    public void testSetThreadTokenThisThread() {
+        HANDLEByReference phToken = new HANDLEByReference();
```

Comment:
I didn't see an extra space persay, but I did notice that both functions had tabs instead of spaces in some areas, so I believe I have fixed those.  I've been struggling a bit with Eclipse's auto spacing.